File: 0036-valid-sudoku/0036-valid-sudoku.java
import java.util.*;
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<Character>[] row = new HashSet[9];
        Set<Character>[] column = new HashSet[9];
        Set<Character>[] box = new HashSet[9];
        
        for(int i = 0; i < 9; i++) {
            row[i] = new HashSet<>();
            column[i] = new HashSet<>();
            box[i] = new HashSet<>();
        }
        
        for(int i = 0; i < 9; i++) {
            for(int j = 0 ; j < 9 ; j++) {
                if(board[i][j] == '.') {
                    continue;
                }
                if(!row[i].add(board[i][j])) {
                    return false;
                }
                if(!column[j].add(board[i][j])) {
                    return false;
                }
                if(!box[(j / 3)* 3 + i / 3].add(board[i][j])){
					return false;
				}
            }
        }
        return true;
        
    }
}

File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        cols = collections.defaultdict(set) # {col : <set of value>}
        rows = collections.defaultdict(set) # {row : <set of value>}
        squares = collections.defaultdict(set) #{(r // 3, c // 3 : <set of values>)
        
        for r in range(9):
            for c in range(9):
                if ( board[r][c] in cols[c] or
                     board[r][c] in rows[r] or
                     board[r][c] in squares[(r // 3, c // 3)]):
                    print(cols)
                    print(rows)
                    print(squares)
                    return False
                if board[r][c] != ".":
                    cols[c].add(board[r][c])
                    rows[r].add(board[r][c])
                    squares[(r // 3,c // 3)].add(board[r][c])
        return True